### DISCUSSION TITLE
How to delete a file from filestorage?

### DISCUSSION THREAD

**Discussion by SamGeens** [vuejs/vuefire#1434] [Questions and Help] [answered]
The [docs](https://vuefire.vuejs.org/api/modules/vuefire.html#useStorageFile) say that it is possible to delete a file with `useStorageFile` however this doesn't seem to return a delete option.

How should a file be deleted?

**Comment by posva** [comment, marked as answer]
There is no directy `delete` method yet. You have to use the firebase SDK instead: https://firebase.google.com/docs/storage/web/delete-files#delete_a_file

**Comment by stildalf** [comment]
Thanks @posva, I been traversing this as well and noticed a bit of a niggle here, seems the `url` & `metadata` Refs don't seem to be updated with `useStorageFile(storageFileRef)` after awaiting a `deleteObject(storageFileRef)` and `refresh()`.

**Reply by posva** [reply]
That's normal, they are not aware. Maybe in the future we add a delete method

**Reply by stildalf** [reply]
Thanks for the clarification, will see if I can submit a PR sometime. I see there's an unexposed and yet unused `_deleteObject()` in `useStorageFile()`.
